extensions/vscode/src/multiStepInputs/newCredential.ts

```diff
@@ -69,6 +69,16 @@ export async function newCredential(
     return state;
   }
 
+  function hasExistingCredential(serverURL: string): boolean {
+    return (
+      credentials.find((credential) => {
+        const existing = normalizeURL(credential.url).toLowerCase();
+        const newURL = normalizeURL(serverURL).toLowerCase();
+        return newURL.includes(existing);
+      }) !== null
+    );
+  }
+
```

Comment:
Would you mind moving this logic into `state.ts`, and call it `findCredentialByURL`? That way, it will sit beside the existing `findCredential` (which operates on name), and be available to use from your `newCredential` code as well. 

I'd suggest also following that same pattern that `findCredential` uses, to simply return null when it can't find it, and then you can evaluate the result against it to answer the question if it already exists.

Reply:
I moved that function to state.ts but imported just the function since it doesn't seem like newCredential and newDeployment are designed to use PublisherState directly. Let me know if there's a better way to do that.

Reply:
>it doesn't seem like newCredential and newDeployment are designed to use PublisherState directly

@jeffvroom , you are absolutely right. I'm sorry for the incorrect direction that I provided you. 

Tomorrow, I'm going to push up some changes to this branch that should satisfy what I was looking for without causing you any additional effort. When I do get a commit pushed up, I'll post another message and if you like those changes, we can get this PR merged. Thanks again for all of your effort on this.

Reply:
Cool, that will make this a lot cleaner. Thanks for your help.